### DISCUSSION TITLE
Replace a Ceph monitor

### DISCUSSION THREAD

**Discussion by jwahsnakupaku** [rook/rook#13712] [Q&A] [answered]
Hi,

I've currently got a 3 node Rook Cluster, I would like to decommission a node and replace it with another.
I've got a reasonable idea of how to handle all the OSDs, my plan is to just add the new node in with OSDs and then down then remove the old ones.
The thing I'm struggling with is what to do with a monitor, there doesn't seem to be a process I can find to decommission a monitor.
Atm I've got mon-a, mon-b, mon-c, ideally I should be able to add the new node, create mon-d and then remove mon-a?

I've cobbled together some steps based on some of the troubleshooting issues, surprised I couldn't find some exact steps in my searching though.
If I've missed a bit of documentation, happy to be pointed in the right direction, but I think I should be able to;
- Change mon: count to 4
- Change maxMonId to 3
- Restart rook-operator
- At this point I should have a mon-d
- Stop rook-operator
- Delete mon-a replicaSet, deployment, pod
- Change mon: count to 3
- Start rook-operator

Is there a simpler way to do this? I shouldn't have to do this often so if the above is expected to work I'll be pretty happy.
Cheers!

**Comment by travisn** [comment, marked as answer]
@jwahsnakupaku The mon failover will be performed automatically by Rook. See the [Monitor Health](https://rook.io/docs/rook/latest/Storage-Configuration/Advanced/ceph-mon-health/#failing-over-a-monitor) topic, it's not the easiest topic to find deep in the docs.

**Reply by travisn** [reply]
After the new node is added, then just scale down the mon on the node you want to remove, and a new mon would be created on the new node and the old mon automatically removed.

**Reply by jwahsnakupaku** [reply]
Cheers for that, worked a charm!
